### PR TITLE
Add docs to `path`.

### PR DESCRIPTION
**Pull request by AndrewSouthpaw**
Addresses discussion and suggestions in #3007 to clarify documentation of `path`.

I'm not sure if any of these doc edits would be welcome, but I thought I'd at least offer them. Feel free to tweak or reject if it doesn't match the style.

### CODE REVIEW

File: source/path.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ import _curry2 from './internal/_curry2';
 import paths from './paths';
 
 /**
- * Retrieve the value at a given path.
+ * Retrieve the value at a given path. String `Idx` that could be parsed as
+ * non-negative integers has the same lookup behavior as their Integer
+ * counterparts. Non-negative integer `Idx` only work on an Array `obj`.
```
Comment (CrossEye) [inline source/path.js:7]: While I agree that this needs some better explanation, I think this suggestion is too tied to your use-case, which I think is fairly unusual.  I'll try to think about it this weekend.

**Comment by CrossEye** [comment]
Such suggestions are always welcome!

In this case, I disagree on the details.  I'll try to think of some more comprehensive text this weekend.

**Comment by AndrewSouthpaw** [comment]
👍  Let me know if there's more I can do to help.

**Comment by CrossEye** [comment]
Yeah, sorry.  I dropped this entirely.  I will look again.

File: source/path.js (hunk 1 above)
Comment (CrossEye) [reply source/path.js:7]: How about something like this?

> Retrieves the value at a given path.  The nodes of the path can be arbitrary strings or non-negative integers.  For anything else, the value is unspecified.  Integer paths are meant to index arrays, strings are meant for objects.

File: source/path.js [hunk 2]
```diff
@@ -11,7 +13,7 @@ import paths from './paths';
  * @typedefn Idx = String | Int
  * @sig [Idx] -> {a} -> a | Undefined
```
Comment (CrossEye) [inline source/path.js:14]: Perhaps this instead?:

```js
/*
 * @sig [Idx] → {a} → a | Undefined
 * @sig Idx = String | NonNegativeInt
 */
```

File: source/path.js [hunk 3]
```diff
@@ -11,7 +13,7 @@ import paths from './paths';
...
  * @param {Array} path The path to use.
- * @param {Object} obj The object to retrieve the nested property from.
+ * @param {Object} obj The object or array to retrieve the nested property from.
  * @return {*} The data at `path`.
  * @see R.prop, R.nth
```
Comment (CrossEye) [inline source/path.js:18]: We should also mention `assocPath`/`dissocPath` here.

**Comment by customcommander** [comment]
@CrossEye I rebased against the master branch and added your suggestions.

File: source/path.js (hunk 2 above)
Comment (customcommander) [reply source/path.js:13]: Done. Can you check please?

File: source/path.js (hunk 3 above)
Comment (customcommander) [reply source/path.js]: Done.

**Review by CrossEye — APPROVED** [review]

**Comment by AndrewSouthpaw** [comment]
Oops I totally forgot about this, thanks for carrying it across the finish line, @customcommander!

**Pull request merged**